src/test/compile-fail/call-fn-never-arg-wrong-type.rs

```diff
@@ -8,8 +8,13 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-fn forever2() -> ! { //~ ERROR computation may converge in a function marked as diverging
-  loop { break }
+#![feature(never_type)]
+
```

Comment:
I've added comments to all of them now. These tests are intended (for the most part) to follow the list you gave earlier.